src/main/java/com/thebois/models/beings/Pawn.java

```diff
@@ -0,0 +1,62 @@
+package com.thebois.models.beings;
+
+import com.thebois.models.Position;
+
+/**
+ * Pawn.
+ */
+public class Pawn implements IBeing {
+
+    private final Position currentPosition;
+    private Position destination;
```

Comment:
And you may wonder how you would unit test this random destination since you can't control the randomness in the test.

The trick is that you don't let the `Pawn` construct its own instance of `Random`, you pass it in the constructor:
```java
public Pawn(Random random) {
    this.random = random;
}
```
and this means that in your unit test you can create a mock random that returns predetermined values:
```java
// Arrange
Random mockRandom = Mockito.mock(Random.class);
when(mockRandom.nextInt()).thenReturn(1);

IBeing pawn = new Pawn(mockRandom);

// Act
pawn.update();

// Assert
assertThat(pawn.getPosition()).isEqualTo(expectedPosition);
```

Reply:
Thanks for the solution, I will keep this in mind!